pandapower/control/util/auxiliary.py

```diff
@@ -210,41 +212,182 @@ def plot_characteristic(characteristic, start, stop, num=20, xlabel=None, ylabel
-        single_mode = False
-        if not (len(trafo_index) == len(x_points) == len(y_points)):
-            raise UserWarning("The lengths of the trafo index and points do not match!")
+def create_trafo_characteristic_object(net):
+    # check if trafo_characteristic_spline table already exists
+    if "trafo_characteristic_spline" in net:
+        raise UserWarning("A trafo_characteristic_spline table already exists - "
+                          "no trafo characteristic objects created.")
+    # 2-winding transformers
+    if (net['trafo_characteristic_table'].index.size > 0 and
+            net['trafo_characteristic_table']['vk_percent'].notna().any()):
+        time_start = time.time()
+        logger.info("Creating tap dependent characteristic objects for 2w-trafos.")
+        characteristic_df_temp = net['trafo_characteristic_table'][
+            ['id_characteristic', 'step', 'voltage_ratio', 'angle_deg', 'vk_percent', 'vkr_percent']]
+        for trafo_id, trafo_row in net.trafo.dropna(subset=['id_characteristic_table']).iterrows():
+            characteristic_df = characteristic_df_temp.loc[
+                characteristic_df_temp['id_characteristic'] == trafo_row['id_characteristic_table']]
+            variables = ['voltage_ratio', 'angle_deg', 'vk_percent', 'vkr_percent']
+            variables_filtered = [var for var in variables if var in characteristic_df.columns]
+            x_points = [characteristic_df['step'].to_list()]
+            y_points = {col: [characteristic_df[col].tolist()] for col in variables_filtered}
+            logger.info("Adding characteristic objects for 2w-trafo with trafo_id %s and variables "
+                        "%s" % (trafo_id, variables_filtered))
+            create_trafo_characteristics(net, "trafo", [trafo_id], variables_filtered,
+                                         x_points, y_points)
+        logger.info(f"Finished creating tap dependent characteristic objects for 2w-trafos in "
+                    f"{time.time() - time_start}.")
     else:
-        single_mode = True
-        if (len(x_points) != len(y_points)):
-            raise UserWarning("The lengths of the points do not match!")
+        logger.info("trafo_characteristic_table has no values for 2w-trafos - no characteristic objects created.")
+    # 3-winding transformers
+    if (net['trafo_characteristic_table'].index.size > 0 and
+            net['trafo_characteristic_table']['vkr_hv_percent'].notna().any()):
+        time_start = time.time()
+        logger.info("Creating tap dependent characteristic objects for 3w-trafos.")
+        characteristic_df_temp = net['trafo_characteristic_table'][
+                ['id_characteristic', 'step', 'voltage_ratio', 'angle_deg', 'vk_hv_percent', 'vkr_hv_percent',
+                 'vk_mv_percent', 'vkr_mv_percent', 'vk_lv_percent', 'vkr_lv_percent']]
+        for trafo_id, trafo_row in net.trafo3w.dropna(subset=['id_characteristic_table']).iterrows():
```

Comment:
checking for deprecation warnings

Reply:
I see no warnings at this point - will re-assess in the future